sdk/src/connections/target/rgb_sensor.cpp

```diff
@@ -0,0 +1,663 @@
+            }
+        } else {
+            /* Error ignored */
+        }
+
+        /* Set the frame format in the driver */
+        CLEAR(fmt);
+        fmt.type = V4L2_BUF_TYPE_VIDEO_CAPTURE;
+        fmt.fmt.pix.width = 1920;
+        fmt.fmt.pix.height = 1080;
+        fmt.fmt.pix.pixelformat = V4L2_PIX_FMT_SRGGB10;
+        fmt.fmt.pix.field = V4L2_FIELD_INTERLACED;
+
+        if (xioctl(dev->fd, VIDIOC_S_FMT, &fmt) == -1) {
+            LOG(WARNING) << "Setting Pixel Format error, errno: " << errno
+                         << " error: " << strerror(errno);
+            return Status::GENERIC_ERROR;
+        }
+
+        /* Buggy driver paranoia. */
+        min = fmt.fmt.pix.width * 2;
+        if (fmt.fmt.pix.bytesperline < min)
+            fmt.fmt.pix.bytesperline = min;
+        min = fmt.fmt.pix.bytesperline * fmt.fmt.pix.height;
+        if (fmt.fmt.pix.sizeimage < min)
+            fmt.fmt.pix.sizeimage = min;
+    }
+
+    return status;
+}
+
+aditof::Status RgbSensor::start() {
+    using namespace aditof;
+    Status status = Status::OK;
+    struct VideoDev *dev;
+    struct v4l2_buffer buf;
+
+    for (unsigned int i = 0; i < m_implData->numVideoDevs; i++) {
+        dev = &m_implData->videoDevs[i];
+        if (dev->started) {
+            LOG(INFO) << "Device already started";
+            return Status::BUSY;
+        }
+        LOG(INFO) << "Starting device " << i;
+
+        for (unsigned int i = 0; i < dev->nVideoBuffers; i++) {
+            CLEAR(buf);
+            buf.type = V4L2_BUF_TYPE_VIDEO_CAPTURE;
+            buf.memory = V4L2_MEMORY_MMAP;
+            buf.index = i;
+
+            if (xioctl(dev->fd, VIDIOC_QBUF, &buf) == -1) {
+                LOG(WARNING)
+                    << "mmap error "
+                    << "errno: " << errno << " error: " << strerror(errno);
+                return Status::GENERIC_ERROR;
+            }
+        }
+        dev->videoBuffersType = V4L2_BUF_TYPE_VIDEO_CAPTURE;
+        if (xioctl(dev->fd, VIDIOC_STREAMON, &dev->videoBuffersType) != 0) {
+            LOG(WARNING) << "VIDIOC_STREAMON error "
+                         << "errno: " << errno << " error: " << strerror(errno);
+            return Status::GENERIC_ERROR;
+        }
+
+        dev->started = true;
+    }
+    return status;
+}
+
+aditof::Status RgbSensor::stop() {
+    using namespace aditof;
+    Status status = Status::OK;
+    struct VideoDev *dev;
+    enum v4l2_buf_type type;
+    struct v4l2_buffer buf;
+
+    for (unsigned int i = 0; i < m_implData->numVideoDevs; i++) {
+        dev = &m_implData->videoDevs[i];
+        type = V4L2_BUF_TYPE_VIDEO_CAPTURE;
+        if (-1 == xioctl(dev->fd, VIDIOC_STREAMOFF, &type))
+            return Status::GENERIC_ERROR;
+    }
+    return status;
+}
+
+aditof::Status
+RgbSensor::getAvailableFrameTypes(std::vector<aditof::FrameDetails> &types) {
+    using namespace aditof;
+    Status status = Status::OK;
+    for (int i = 0; i < types.size(); i++) {
```

Comment:
types is an output parameter. This means that this function will set a value to it and not rely on it's old value. You don't need a for here. There is only one type that you need to add which is called: "rgb".

Reply:
As for the last week's discussion I was told that rgb camera shouldn't have a separate frame type, rather try to combine the already existing types with rgb, since we use them in parrallel. The following options were given: 

- depth_only-rgb
- ir_only-rgb
- depth_ir-rgb
 in order to combine the naming I just had to add to the already existing frametypes the new naming

Reply:
Things discussed are still true. But the combined frame type belong to the camera. These changes are in rgb sensor not in the camera. The sensor currently has one type and that is rgb.